### ISSUE TITLE
ast/non_null_attr

### ISSUE THREAD

**Issue by Konstantin8105** [elliotchance/c2go#361]
I don't understood - The reason of fail. I do not update clang and we do not update version of SQLite.
```
----------------------
Transpiling shell.c...
panic: could not match regexp with string
^(?P<address>[0-9a-fx]+) <(?P<position>.*)>(?P<a> \d+)(?P<b> \d+)?(?P<c> \d+)?[\s]*$
0x248ea60 <line:155:26, col:49> 0 1 2 4


goroutine 33 [running]:
github.com/elliotchance/c2go/ast.groupsFromRegex(0xc42104bd40, 0x54, 0xc421320d7a, 0x27, 0x200000000040f14f)
	/home/konstantin/go/src/github.com/elliotchance/c2go/ast/ast.go:250 +0x3cc
github.com/elliotchance/c2go/ast.parseNonNullAttr(0xc421320d7a, 0x27, 0x5e2deb)
	/home/konstantin/go/src/github.com/elliotchance/c2go/ast/non_null_attr.go:22 +0x62
github.com/elliotchance/c2go/ast.Parse(0xc421320d6e, 0x33, 0x5e16bf, 0x5)
	/home/konstantin/go/src/github.com/elliotchance/c2go/ast/ast.go:166 +0x25c8
main.convertLinesToNodes(0xc4201a2000, 0x2d02, 0xb409, 0x0, 0x0, 0x0)
	/home/konstantin/go/src/github.com/elliotchance/c2go/main.go:96 +0x196
main.convertLinesToNodesParallel.func1.1(0xc421748000, 0xc42000c008, 0xc4201a2000, 0x2d02, 0xb409, 0x0)
	/home/konstantin/go/src/github.com/elliotchance/c2go/main.go:120 +0x59
created by main.convertLinesToNodesParallel.func1
	/home/konstantin/go/src/github.com/elliotchance/c2go/main.go:124 +0x129
```
Clang:
```
$ clang --version
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

**Comment by elliotchance** [comment]
I'm not sure why that failed either. It is fixed when you add that test case for that regex?

**Comment by yulvil** [comment]
Most likely caused by the regexp changes I made. I will have a look


On Nov 23, 2017, at 14:48, Elliot Chance <notifications@github.com> wrote:

I'm not sure why that failed either. It is fixed when you add that test case for that regex?

**Comment by Konstantin8105** [comment]
@yulvil  I look on your [PR](https://github.com/elliotchance/c2go/commit/be4fee1183cf0208becd1ef4c49210e498fffca0#diff-9eb9b4209616e8a865f062d03ce47fe3) and your changes in file `ast/non_null_attr.go` is look good.
Could you clarify changes in file ` ast/ast.go` : 
```go
rx = fullRegexp + "[\\s]*$" // <-- Added plus ....
```
In according to https://golang.org/pkg/regexp/syntax/ , it is only ignored any `whitespaces` at the end on line.
So, all is great and my `clang` have some specific.

**Comment by Konstantin8105** [comment]
Dear @yulvil and @elliotchance 
If we look [clang::NonNullAttr Member List](https://www.crest.iu.edu/projects/conceptcpp/docs/html-ext/classclang_1_1NonNullAttr-members.html), then we will see full list of members. Like I see, NonNullAttr have attribute `Inherited` like all `InheritableAttr` elements, but we don't implemented in `c2go.ast`. 

In according to [diagramm](https://www.crest.iu.edu/projects/conceptcpp/docs/html-ext/classclang_1_1Attr.html), `ConstAttr ` also attribute `Inherited` like all `InheritableAttr` elements, but we don't implemented in `c2go.ast`. 

Have you idea - how to easy parse?

**Comment by elliotchance** [comment]
There are many part of the clang AST that are not implemented since we are only creating test cases when the issues arise. It would not be worth trying to implement everything else for a few reasons:

1. The attributes may not be used anywhere, ever.
2. It creates a lot of extra code that we may never use (lowering the code coverage).
3. It could only be triggered in environments we done use (like only with C++ code).

We should continue to write test cases as the syntax is discovered.

**Comment by Konstantin8105** [comment]
solved #478
